vars/deployProd.groovy

```diff
@@ -35,7 +35,46 @@ def call(body) {
 
 
   stage('Checkout Tag') {
-    sh "git checkout tags/${config.prodVersion}"
+    sh "git checkout master"
+  }
+
+  def url = ''
+  def urlMinusProtocol = ''
+  withCredentials([usernamePassword(credentialsId: 'github', passwordVariable: 'GIT_PASSWORD', usernameVariable: 'GIT_USERNAME')]) {
+      stage('Check master branch') {
+          url = sh(returnStdout: true, script: 'git config remote.origin.url').trim()
+          urlMinusProtocol = url.substring(url.indexOf('://')+3)
+
+          sh "git fetch --no-tags --progress https://${GIT_USERNAME}:${GIT_PASSWORD}@${urlMinusProtocol} +refs/tags/${config.prodVersion}:refs/remotes/origin/master"
+
+          //Compare to master branch to look for any unmerged changes in the tag
```

Comment:
After thinking about this some more, this check is always going to fail if we are deploying an older version to prod (for example rolling back a bad deployment). 
I think we need to change this comparison to check to see if the tag has changes that aren't in master and if it can be merged without conflicts. If that is the case, then we should do the merge to master. If there are changes, but conflicts then we should fail the deploy.
If there are no changes on the tag that aren't in master, then we can skip the merge and just perform the release. I also think we should do the release from the tag branch instead of the master branch which is what is happening now.